### PR TITLE
All ansible roles

### PR DESCRIPTION
**Pull request by sbesson**
While working on https://github.com/openmicroscopy/ansible-role-anonymous-ftp/pull/1, I realized this role was not in the list. This is primarily due to the fact the registration script only returns the 50 first items while we have more than 50 Ansible roles.

This PR update the script to handle the GitHub API pagination and match the ome-ansible installation step in `.travis.yml`. It also adds a `git remote update --remote` step to update the versions.

To test this PR, check Travis is green and that all known Ansible roles are included.

### CODE REVIEW

File: scripts/register_ansible_roles.py [hunk 1]
```diff
@@ -36,31 +35,46 @@
...
     "ansible-role-haproxy": "Uses a non-standard test from upstream",
     "ansible-role-munin-node":
         "No molecule.yml or test.yml (tested by munin role)",
     "ansible-role-omero-logmonitor": "Molecule test doesn't work",
     "ansible-role-omero-web-apps": "Broken (deprecated?)",
-    "ansible-role-devspace": "",
-    "ansible-role-docker": "",
-    "ansible-role-celery-docker": "",
+    "ansible-role-devspace": "Docker/docker",
+    "ansible-role-docker": "Docker/docker",
```
Comment (manics) [inline scripts/register_ansible_roles.py]: What's failing here? https://travis-ci.org/openmicroscopy/ansible-role-docker looks fine

**Comment by sbesson** [comment]
@manics reincluded but see https://travis-ci.org/ome/ansible-roles/jobs/390792869. For https://github.com/ome/ansible-roles/pull/1#pullrequestreview-127514394, would be inclined to test in a follow-up branch/PR and have an issue collecting all the roles that need reviewing (mindful of the overarching plan which is to move to Ansible 2.4/Molecule 2 anyways)

**Comment by manics** [comment]
Looks like there's bee a recent change to the distribution repos which means docker 1.12.6 is no longer published `No package matching 'docker-1.12.6' found available, installed or updated`

**Comment by manics** [comment]
It took me a while to figure out how to use `register_ansible_roles.py`, I could only run it in an empty repo, not an existing one:
```
$ git init ansible-roles-new
$ cd ansible-roles-new/
$ python ../scripts/register_ansible_roles.py
```
```
$ git status
On branch master

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)

        new file:   .gitmodules
        new file:   ansible-role-analysis-tools
        new file:   ansible-role-anonymous-ftp
        new file:   ansible-role-basedeps
        new file:   ansible-role-cadvisor
        new file:   ansible-role-celery-docker
        new file:   ansible-role-cli-utils
        new file:   ansible-role-debug-dumpallvars
        new file:   ansible-role-devspace
        new file:   ansible-role-docker
        new file:   ansible-role-docker-slack-notifier
        new file:   ansible-role-docker-tools
        new file:   ansible-role-fluentd
        new file:   ansible-role-hosts-populate
        new file:   ansible-role-ice
        new file:   ansible-role-iptables-raw
        new file:   ansible-role-java
        new file:   ansible-role-jekyll-build
        new file:   ansible-role-local-accounts
        new file:   ansible-role-logrotate
        new file:   ansible-role-lvm-partition
        new file:   ansible-role-mysql-backup
        new file:   ansible-role-network
        new file:   ansible-role-network-cloud-interfaces
        new file:   ansible-role-nfs-mount
        new file:   ansible-role-nfs-share
        new file:   ansible-role-nginx
        new file:   ansible-role-nginx-proxy
        new file:   ansible-role-nginx-ssl-selfsigned
        new file:   ansible-role-omego
        new file:   ansible-role-omero-common
        new file:   ansible-role-omero-logmonitor
        new file:   ansible-role-omero-prometheus-exporter
        new file:   ansible-role-omero-python-deps
        new file:   ansible-role-omero-server
        new file:   ansible-role-omero-user
        new file:   ansible-role-omero-web
        new file:   ansible-role-omero-web-apps
        new file:   ansible-role-omero-web-django-prometheus
        new file:   ansible-role-omero-web-runtime
        new file:   ansible-role-openstack-volume-storage
        new file:   ansible-role-postgresql
        new file:   ansible-role-prometheus
        new file:   ansible-role-prometheus-jmx
        new file:   ansible-role-prometheus-node
        new file:   ansible-role-prometheus-postgres
        new file:   ansible-role-python-pydata
        new file:   ansible-role-python-virtualenv-upstream
        new file:   ansible-role-reboot-server
        new file:   ansible-role-redis
        new file:   ansible-role-rsync-server
        new file:   ansible-role-samba-client
        new file:   ansible-role-selinux-utils
        new file:   ansible-role-ssl-certificate
        new file:   ansible-role-storage-volume-initialise
        new file:   ansible-role-sudoers
        new file:   ansible-role-system-monitor-agent
        new file:   ansible-role-upgrade-distpackages
        new file:   ansible-role-versioncontrol-utils
        new file:   infrastructure-extract-ansible-role
        new file:   ome-ansible-molecule-dependencies

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        .travis.yml

$ diff ../.travis.yml .travis.yml
# (No differences)
```
A top-level README describing the purpose of this repo is one obvious missing item, but can be left for a later PR if you prefer

**Comment by sbesson** [comment]
:+1: for a readme as a follow-up. What error did you get with `register_ansible_roles.py` I certainly used it for generating many of the commits in this PR so the expectation is definitely that it should be usable within the repository (and maybe renamed as `update_ansible_roles.py`)

**Comment by manics** [comment]
```
$ git clone https://github.com/ome/ansible-roles.git
$ cd ansible-roles/
# <Fetch this PR from github>
$ git checkout pr1
$ $ python scripts/register_ansible_roles.py
'ome-ansible-molecule-dependencies' already exists in the index
'ansible-role-analysis-tools' already exists in the index
Submodule path 'ansible-role-analysis-tools' not initialized
Maybe you want to use 'update --init'?
'ansible-role-anonymous-ftp' already exists in the index
Submodule path 'ansible-role-anonymous-ftp' not initialized
Maybe you want to use 'update --init'?
'ansible-role-basedeps' already exists in the index
Submodule path 'ansible-role-basedeps' not initialized
Maybe you want to use 'update --init'?
'ansible-role-cadvisor' already exists in the index
Submodule path 'ansible-role-cadvisor' not initialized
Maybe you want to use 'update --init'?
'ansible-role-celery-docker' already exists in the index
Submodule path 'ansible-role-celery-docker' not initialized
Maybe you want to use 'update --init'?
'ansible-role-cli-utils' already exists in the index
Submodule path 'ansible-role-cli-utils' not initialized
Maybe you want to use 'update --init'?
'ansible-role-debug-dumpallvars' already exists in the index
Submodule path 'ansible-role-debug-dumpallvars' not initialized
Maybe you want to use 'update --init'?
'ansible-role-devspace' already exists in the index
Submodule path 'ansible-role-devspace' not initialized
Maybe you want to use 'update --init'?
'ansible-role-docker-slack-notifier' already exists in the index
Submodule path 'ansible-role-docker-slack-notifier' not initialized
Maybe you want to use 'update --init'?
'ansible-role-docker-tools' already exists in the index
Submodule path 'ansible-role-docker-tools' not initialized
Maybe you want to use 'update --init'?
'ansible-role-docker' already exists in the index
Submodule path 'ansible-role-docker' not initialized
Maybe you want to use 'update --init'?
'ansible-role-fluentd' already exists in the index
Submodule path 'ansible-role-fluentd' not initialized
Maybe you want to use 'update --init'?
'ansible-role-hosts-populate' already exists in the index
Submodule path 'ansible-role-hosts-populate' not initialized
Maybe you want to use 'update --init'?
'ansible-role-ice' already exists in the index
Submodule path 'ansible-role-ice' not initialized
Maybe you want to use 'update --init'?
'ansible-role-iptables-raw' already exists in the index
Submodule path 'ansible-role-iptables-raw' not initialized
Maybe you want to use 'update --init'?
'ansible-role-java' already exists in the index
Submodule path 'ansible-role-java' not initialized
Maybe you want to use 'update --init'?
'ansible-role-jekyll-build' already exists in the index
Submodule path 'ansible-role-jekyll-build' not initialized
Maybe you want to use 'update --init'?
'ansible-role-local-accounts' already exists in the index
Submodule path 'ansible-role-local-accounts' not initialized
Maybe you want to use 'update --init'?
'ansible-role-logrotate' already exists in the index
Submodule path 'ansible-role-logrotate' not initialized
Maybe you want to use 'update --init'?
'ansible-role-lvm-partition' already exists in the index
Submodule path 'ansible-role-lvm-partition' not initialized
Maybe you want to use 'update --init'?
'ansible-role-mysql-backup' already exists in the index
Submodule path 'ansible-role-mysql-backup' not initialized
Maybe you want to use 'update --init'?
'ansible-role-network-cloud-interfaces' already exists in the index
Submodule path 'ansible-role-network-cloud-interfaces' not initialized
Maybe you want to use 'update --init'?
'ansible-role-network' already exists in the index
Submodule path 'ansible-role-network' not initialized
Maybe you want to use 'update --init'?
'ansible-role-nfs-mount' already exists in the index
Submodule path 'ansible-role-nfs-mount' not initialized
Maybe you want to use 'update --init'?
'ansible-role-nfs-share' already exists in the index
Submodule path 'ansible-role-nfs-share' not initialized
Maybe you want to use 'update --init'?
'ansible-role-nginx-proxy' already exists in the index
Submodule path 'ansible-role-nginx-proxy' not initialized
Maybe you want to use 'update --init'?
'ansible-role-nginx-ssl-selfsigned' already exists in the index
Submodule path 'ansible-role-nginx-ssl-selfsigned' not initialized
Maybe you want to use 'update --init'?
'ansible-role-nginx' already exists in the index
Submodule path 'ansible-role-nginx' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omego' already exists in the index
Submodule path 'ansible-role-omego' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-common' already exists in the index
Submodule path 'ansible-role-omero-common' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-logmonitor' already exists in the index
Submodule path 'ansible-role-omero-logmonitor' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-prometheus-exporter' already exists in the index
Submodule path 'ansible-role-omero-prometheus-exporter' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-python-deps' already exists in the index
Submodule path 'ansible-role-omero-python-deps' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-server' already exists in the index
Submodule path 'ansible-role-omero-server' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-user' already exists in the index
Submodule path 'ansible-role-omero-user' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-web-apps' already exists in the index
Submodule path 'ansible-role-omero-web-apps' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-web-django-prometheus' already exists in the index
Submodule path 'ansible-role-omero-web-django-prometheus' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-web-runtime' already exists in the index
Submodule path 'ansible-role-omero-web-runtime' not initialized
Maybe you want to use 'update --init'?
'ansible-role-omero-web' already exists in the index
Submodule path 'ansible-role-omero-web' not initialized
Maybe you want to use 'update --init'?
'ansible-role-openstack-volume-storage' already exists in the index
Submodule path 'ansible-role-openstack-volume-storage' not initialized
Maybe you want to use 'update --init'?
'ansible-role-postgresql' already exists in the index
Submodule path 'ansible-role-postgresql' not initialized
Maybe you want to use 'update --init'?
'ansible-role-prometheus-jmx' already exists in the index
Submodule path 'ansible-role-prometheus-jmx' not initialized
Maybe you want to use 'update --init'?
'ansible-role-prometheus-node' already exists in the index
Submodule path 'ansible-role-prometheus-node' not initialized
Maybe you want to use 'update --init'?
'ansible-role-prometheus-postgres' already exists in the index
Submodule path 'ansible-role-prometheus-postgres' not initialized
Maybe you want to use 'update --init'?
'ansible-role-prometheus' already exists in the index
Submodule path 'ansible-role-prometheus' not initialized
Maybe you want to use 'update --init'?
'ansible-role-python-pydata' already exists in the index
Submodule path 'ansible-role-python-pydata' not initialized
Maybe you want to use 'update --init'?
'ansible-role-python-virtualenv-upstream' already exists in the index
Submodule path 'ansible-role-python-virtualenv-upstream' not initialized
Maybe you want to use 'update --init'?
'ansible-role-reboot-server' already exists in the index
Submodule path 'ansible-role-reboot-server' not initialized
Maybe you want to use 'update --init'?
'ansible-role-redis' already exists in the index
Submodule path 'ansible-role-redis' not initialized
Maybe you want to use 'update --init'?
'ansible-role-rsync-server' already exists in the index
Submodule path 'ansible-role-rsync-server' not initialized
Maybe you want to use 'update --init'?
'ansible-role-samba-client' already exists in the index
Submodule path 'ansible-role-samba-client' not initialized
Maybe you want to use 'update --init'?
'ansible-role-selinux-utils' already exists in the index
Submodule path 'ansible-role-selinux-utils' not initialized
Maybe you want to use 'update --init'?
'ansible-role-ssl-certificate' already exists in the index
Submodule path 'ansible-role-ssl-certificate' not initialized
Maybe you want to use 'update --init'?
'ansible-role-storage-volume-initialise' already exists in the index
Submodule path 'ansible-role-storage-volume-initialise' not initialized
Maybe you want to use 'update --init'?
'ansible-role-sudoers' already exists in the index
Submodule path 'ansible-role-sudoers' not initialized
Maybe you want to use 'update --init'?
'ansible-role-system-monitor-agent' already exists in the index
Submodule path 'ansible-role-system-monitor-agent' not initialized
Maybe you want to use 'update --init'?
'ansible-role-upgrade-distpackages' already exists in the index
Submodule path 'ansible-role-upgrade-distpackages' not initialized
Maybe you want to use 'update --init'?
'ansible-role-versioncontrol-utils' already exists in the index
Submodule path 'ansible-role-versioncontrol-utils' not initialized
Maybe you want to use 'update --init'?
'infrastructure-extract-ansible-role' already exists in the index
Submodule path 'infrastructure-extract-ansible-role' not initialized
Maybe you want to use 'update --init'?
'ome-ansible-molecule-dependencies' already exists in the index
Submodule path 'ome-ansible-molecule-dependencies' not initialized
Maybe you want to use 'update --init'?
```

**Comment by sbesson** [comment]
Ah right, you need to recursively initialize all submodules in order to be able to run the script updating the commits. I could also update the script so that `git submodule update --init` is called or call this a documentation/README issue i.e. you need to `git clone && git submodule update --init` before using the script.

**Comment by manics** [comment]
Either solution is fine, as long as it's clear how to run it.

**Comment by sbesson** [comment]
@manics last commits are adding a README and updating the script (now called `update.py` so that it can be run from a fresh repository with or without initialized submodules)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+[![Build Status](https://travis-ci.org/ome/ansible-roles.svg)](https://travis-ci.org/openmicroscopy/ansible-roles)
+
+
+A super-repository collecting of all the existing OME Ansible roles including
+those released on [Galaxy](http://galaxy.ansible.com/openmicroscopy/).
+
+Update
+------
+
+The [update.py](scripts/update.py) allows to add new Ansible roles as
```
Comment (manics) [inline README.md]: The [update.py](scripts/update.py) >**_script_**< allows

**Pull request merged**